Don't override MacPorts-specified CC or CFLAGS.
Has been reported to the developers by email.
--- Makefile.orig	2021-03-07 22:40:27.000000000 -0600
+++ Makefile	2022-03-22 14:08:12.000000000 -0500
@@ -3,8 +3,8 @@
 # (C) 1997-2007 Nelson Murilo, Pangeia Informatica, AMS Foundation and others.
 #
 
-CC       = cc
-CFLAGS	 = -DHAVE_LASTLOG_H
+CC       ?= cc
+CFLAGS	 += -DHAVE_LASTLOG_H
 STATIC   = -static
 
 ###
